lib/nameable.rb

```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
```

Comment:
[OPTIONAL] You can add an error message that is more descriptive of the error.

i.e
```
raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
```